### PR TITLE
Add Filling form POM, Types and BDD test

### PR DESCRIPTION
**Pull request by Sanberqui18**
Tests Passed:
Screen Shot 2024-01-03 at 1 43 18 PM

### CODE REVIEW

File: cypress/e2e/filling.form.cy.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+const formPage = new FormPage();
+const confirmationForm = new ConfirmationForm();
+
+describe("Filling Form", () => {
+  describe("Given the user visits the Demo QA Practice form page", () => {
+    beforeEach(() => {
+      cy.visit("https://demoqa.com/automation-practice-form");
+    });
+
+    describe("When the user fills out all the form and click on the submit button", () => {
```
Comment (salorrego) [inline cypress/e2e/filling.form.cy.ts]:
```suggestion
    describe("When the user fills out all the form and clicks on the submit button", () => {
```

File: cypress/e2e/filling.form.cy.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        formPage.fillEmail(personalInformation.email);
+        formPage.checkGender(personalInformation.gender);
+        formPage.fillDOB(personalInformation.dateOfBirth);
+        formPage.fillPhoneNumber(personalInformation.mobileNumber);
+        formPage.checkHobbies(personalInformation.hobbies);
+        formPage.fillAddress(personalInformation.currentAddress);
+        formPage.submitForm();
+      });
+
+      it("Then the user is rediected to the confirmation page and the data is correct", () => {
```
Comment (salorrego) [inline cypress/e2e/filling.form.cy.ts]:
```suggestion
      it("Then the user should be redirected to the confirmation page and the displayed data should be correct", () => {
```

File: cypress/e2e/filling.form.cy.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          .should(
+            "contain.text",
+            `${personalInformation.hobbies[0]}, ${personalInformation.hobbies[1]}`,
+          );
+        confirmationForm
+          .getValueOf("Address")
+          .should("have.text", personalInformation.currentAddress);
+        confirmationForm.getValueOf("State and City").should("be.empty");
+
+        confirmationForm.closePopUp();
```
Comment (salorrego) [inline cypress/e2e/filling.form.cy.ts:70]: why this action?

File: cypress/pages/confirmation.form.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  private title: string;
+  private table: string;
+  private closeButton: string;
+
+  constructor() {
+    this.title = "#example-modal-sizes-title-lg";
+    this.table = ".table-responsive";
+    this.closeButton = "#closeLargeModal";
+  }
+  getTitle(): Cypress.Chainable<JQuery<HTMLElement>> {
```
Comment (salorrego) [inline cypress/pages/confirmation.form.ts:18]: awesome typing

File: cypress/pages/confirmation.form.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    return cy.get(this.title);
+  }
+
+  getValueOf(filedName: string): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.get(this.table).contains(filedName).next();
+  }
+
+  closePopUp() {
```
Comment (salorrego) [inline cypress/pages/confirmation.form.ts]:
```suggestion
  closePopUp(): void {
```

File: cypress/pages/form.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  fillDOB(DOB: string): void {
+    cy.get(this.dateOfBirthInput).type(DOB);
+  }
+
+  checkHobbies(hobbies: HoobbieList): void {
+    for (let hobby of hobbies) {
+      cy.get(this.hobbies)
+        //@ts-ignore
```
Comment (salorrego) [inline cypress/pages/form.ts]: ??

File: cypress/support/commands.ts [hunk 7]
```diff
@@ -23,3 +23,7 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+
+Cypress.on("uncaught:exception", () => {
+  return false;
+});
```
Comment (salorrego) [inline cypress/support/commands.ts:29]: why?
this probably means you have a problem with the tests

**Review by salorrego — CHANGES_REQUESTED** [review]

File: cypress/e2e/filling.form.cy.ts (hunk 1 above)
Comment (Sanberqui18) [reply cypress/e2e/filling.form.cy.ts]: done

**Review by salorrego — APPROVED** [review]
awesome

**Pull request merged**